packages/cli/src/commands/init/init.ts

```diff
@@ -384,12 +387,62 @@ function checkPackageManagerAvailability(
   return false;
 }
 
-function createTemplateUri(options: Options, version: string): string {
+async function createTemplateUri(
+  options: Options,
+  version: string,
+): Promise<string> {
   const isTypescriptTemplate =
-    options.template === 'react-native-template-typescript';
+    options.template === TEMPLATE_PACKAGE_LEGACY_TYPESCRIPT;
 
   // This allows to correctly retrieve template uri for out of tree platforms.
-  const platform = options.platformName || 'react-native';
+  const platform = options.platformName ?? TEMPLATE_PACKAGE_COMMUNITY;
+
+  let npmVersion: string | undefined;
+
+  if (platform === TEMPLATE_PACKAGE_COMMUNITY) {
```

Comment:
Q: The Template Versioning still needs to be clarified.
If `version` here is `0.75.0-rc.1`, what are you going to resolve on NPM?

As `https://regsitry.npmjs.org/@react-native-community/template/0.75.0-rc.1` will return 404 most likely

We need the version mapping logic to be included here

Reply:
Good callout.

Reply:
I've done a bunch more work on this as well as getting some feedback  in #2345. 
- The template version and react-version can be set independently on `init`.
- By default they'll both select `latest`.
- If you know best, you can avoid any of the measures I've put in and override with `--template <whatever your template is>@<whatever version you want>`. 
- `--template-version` is now a shortcut for `--template @react-native-community/template@<version>`
- `--version` is a shortcut specifically for the version of react-native.

If the new template isn't available, the behaviour is exactly was it was before.